lib/ansible/modules/system/aix_lvg.py

```diff
@@ -0,0 +1,275 @@
+    :return: [True, message]
+    """
+
+    lspv_cmd = module.get_bin_path('lspv', True)
+    rc, current_lspv, err = module.run_command('%s ' % lspv_cmd)
+    if rc != 0:
+        module.fail_json(msg="Failed executing lspv command.", rc=rc,
+                         err=err)
+
+    # check if PV is not already for another volume group
+    for line in current_lspv.splitlines():
+        if line.startswith(pvname):
+            if line.split()[-1] != 'None':
+                module.fail_json(msg="Physical Volume %s already in use." %
+                                     pvname)
+
+    lquerypv_cmd = module.get_bin_path('lquerypv', True)
+    rc, current_lquerypv, err = module.run_command('%s -h /dev/%s 20 10' %
+                                                   (lquerypv_cmd, pvname))
+    if rc != 0:
+        module.fail_json(msg="Failed executing lquerypv command.", rc=rc,
+                         err=err)
+
+    # check if PV is not already in use by Oracle ASM
+    if "ORCLDISK" in current_lquerypv:
+        msg = "PV %s is already used by Oracle ASM" % pvname
+
+        return False, msg
+
+    msg = "PV is ok to be used"
+
+    return True, msg
+
+
+def create_extend_vg(module, vg, pvs, pp_size, vg_type, force):
+    """ creates or extend a volume group"""
+
+    # validate if PV are not already in use
+    for pv in pvs:
+        pv_test_result = _validate_pv(module, pv)
+        if not pv_test_result[0]:
+            module.fail_json(msg=pv_test_result[1])
+
+    # perform action creation or extension
+    # if vg doesn't exist, it will create
+    # if vg already exists, it will try to extend using the PVs
+    lsvg_cmd = module.get_bin_path('lsvg', True)
+    rc, current_vgs, err = module.run_command('%s' % lsvg_cmd)
+    if rc != 0:
+        module.fail_json(msg='Failed executing %s command' % lsvg_cmd)
+
+    if vg in current_vgs:
+        # volume group extension
+        extendvg_cmd = module.get_bin_path('extendvg', True)
+        rc, _, err = module.run_command('%s %s %s' % (extendvg_cmd, vg,
+                                                      ' '.join(pvs)))
+        if rc == 0:
+            module.exit_json(changed=True, msg="Volume group %s extended."
+                                               % vg)
+        else:
+            module.exit_json(changed=False, msg='Extending volume group %s '
+                                                'has failed'
+                                                % vg, rc=rc, err=err)
+
+    else:
+        # volume group creation
+        mkvg_cmd = module.get_bin_path('mkvg', True)
+        rc, _, err = module.run_command('%s %s -s %s %s -y %s %s'
+                                        % (mkvg_cmd, vg_mode[vg_type], pp_size,
+                                           force_mode[force], vg, ' '.join(pvs)
+                                           ))
+        if rc == 0:
+            module.exit_json(changed=True, msg="Volume group %s created" % vg)
```

Comment:
Please could you change the logic the way that this function returns the changed value and you exit at the end of the `main()` function? The same in for the `reduce_vg()` function.